Profile_Properties_Definition.md

```diff
@@ -0,0 +1,246 @@
+		// * Adaptive: Allows implementation to switch between synchronous
+		//   and asynchronous modes.
+		// * Asynchronous: Asynchronous updates.
+		// * Synchronous: Synchronous updates.
+		ReplicaUpdateMode string `json:"replicaUpdateMode,omitempty"`
+		// ConsistencyEnabled indicates that the source and target shall be
+		// consistent. The default value is false.
+		ConsistencyEnabled bool `json:"consistencyEnabled,omitempty"`
+		// ReplicationPeriod shall be an ISO 8601 duration that defines the
+		// duration of performing replication operation. For example,
+		// "P3Y6M4DT12H30M5S" represents a duration of "3 years, 6 months,
+		// 4 days, 12 hours, 30 minutes and 5 seconds".
+		ReplicationPeriod string `json:"replicationPeriod,omitempty"`
+		// ReplicationBandwidth specifies the maximum bandwidth for performing
+		// replication operation.
+		// +units:Mb/s
+		ReplicationBandwidth int64 `json:"replicationBandwidth,omitempty"`
+	} `json:"replicaInfos,omitempty"`
+}
+```
+
+* SnapshotPropertiesSpec
+```go
+type SnapshotPropertiesSpec struct {
+	// The property defines how to take snapshots.
+	Schedule struct {
+		// This vaule is represented as a string in ISO 8601 datetime format.
+		// ISO 8601 sets out an internationally agreed way to represent dates:
+		// yyyy-mm-ddThh:mm:ss.ffffff. For example, "3:53pm, September 15, 2008"
+		// is represented as "2008-09-15T15:53:00".
+		Datetime string `json:"datetime,omitempty"`
+		// The value specifies the duration of executing a operation, which
+		// contains three options including Daily, Weekly and Monthly.
+		Occurrence string `json:"occurrence,omitempty"`
+	} `json:"schedule,omitempty"`
+	Retention struct {
+		// The value specifies the number of recently created snapshots for
+		// retention.
+		// +optional
+		Number int64 `json:"number,omitempty"`
+		// The value specifies the duration of snapshots for retention.
+		// +optional
+		// +units:day
+		Duration int64 `json:"duration,omitempty"`
+	} `json:"retention,omitempty"`
+}
+```
+
+* DataProtectionPropertiesSpec
+```go
+type DataProtectionPropertiesSpec struct {
+	// DataProtection represents some suggested data protection capabilities.
+	DataProtection DataProtectionLoS `json:"dataProtection,omitempty"`
+	// ConsistencyEnabled indicates that the source and target shall be
+	// consistent. The default value is false.
+	ConsistencyEnabled bool `json:"consistencyEnabled,omitempty"`
+}
+```
+
+* CustomPropertiesSpec
+```go
+// CustomPropertiesSpec is a dictionary object that contains unique keys and
+// JSON objects.
+type CustomPropertiesSpec map[string]interface{}
+```
+
+* ProfileSpec
+```go
+// An OpenSDS profile is identified by a unique name and ID. With adding
+// profile properties, each profilecontains a set of tags of storage
+// capabilities which are desirable features for a class of applications.
+type ProfileSpec struct {
+	*BaseModel
+	// The uuid of project
+	// + readOnly
+	TenantId string `json:"tenantId"`
```

Comment:
I didn't notice it before, but sometimes it does exists that some profiles are exposed to specific tenants.

Reply:
Please remove TenantId from ProfileSpec because we only support public profiles now.  We can consider adding support for private profiles in the future.